main.go

```diff
@@ -168,6 +168,12 @@ func main() {
 
 	setupLog.V(1).Info(fmt.Sprintf("feature gates: %+v\n", feature.Gates))
 
+	externalResourceGC := false
+	if feature.Gates.Enabled(feature.ExternalResourceGC) {
+		setupLog.Info("enabling external resource garbage collection")
+		externalResourceGC = true
+	}
+
```

Comment:
I think this belongs in the `enableGates` function, but I understand why it does not, at the moment: We need do this before we initialize AWSClusterReconciler, which we do before we call `enableGates`.